### PR TITLE
Enable nullable reference types for Emby.Photos and Emby.Notifications

### PR DESCRIPTION
**Pull request by Bond-009**
* Enable TreatWarningsAsErrors for Emby.Notifications
* Add analyzers to Emby.Notifications
* ~~**Remove non functioning API endpoints (/Notifications/{UserId}, /Notifications/{UserId}/Summary, /Notifications/{UserId}/Read, /Notifications/{UserId}/UnRead) (Possibly requires changes in clients)**~~

~~Kind of depends on #2136 to disable a warnings in the ruleset~~

### CODE REVIEW

File: Emby.Notifications/NotificationManager.cs [hunk 1]
```diff
@@ -16,33 +16,47 @@
...
-
-        public NotificationManager(ILoggerFactory loggerFactory, IUserManager userManager, IServerConfigurationManager config)
+        private INotificationService[] _services = Array.Empty<INotificationService>();
+        private INotificationTypeFactory[] _typeFactories = Array.Empty<INotificationTypeFactory>();
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="NotificationManager" /> class.
+        /// </summary>
+        /// <param name="logger">The logger.</param>
+        /// <param name="userManager">the user manager.</param>
```
Comment (dkanada) [inline Emby.Notifications/NotificationManager.cs]:
```suggestion
        /// <param name="userManager">The user manager.</param>
```

File: Emby.Notifications/Api/NotificationsService.cs [hunk 2]
```diff
@@ -127,32 +135,29 @@ public NotificationsService(INotificationManager notificationManager, IUserManag
...
             };
         }
 
         public Task Post(AddAdminNotification request)
         {
             // This endpoint really just exists as post of a real with sickbeard
-            return AddNotification(request);
-        }
-
-        private Task AddNotification(AddAdminNotification request)
```
Comment (dkanada) [inline Emby.Notifications/Api/NotificationsService.cs]: Did you mean to remove this method? I agree, just checking.

File: Emby.Notifications/NotificationEntryPoint.cs [hunk 3]
```diff
@@ -269,23 +290,37 @@ private async Task SendNotification(NotificationRequest notification, BaseItem r
...
-            _appHost.HasPendingRestartChanged -= _appHost_HasPendingRestartChanged;
-            _appHost.HasUpdateAvailableChanged -= _appHost_HasUpdateAvailableChanged;
-            _activityManager.EntryCreated -= _activityManager_EntryCreated;
+            Dispose(true);
+            GC.SuppressFinalize(this);
         }
 
-        private void DisposeLibraryUpdateTimer()
+        /// <summary>
+        /// Releases unmanaged and - optionally - managed resources.
```
Comment (dkanada) [inline Emby.Notifications/NotificationEntryPoint.cs]:
```suggestion
        /// Releases unmanaged and optionally managed resources.
```

**Comment by dkanada** [comment]
It also seems like the CI jobs are failing, so that also needs to be fixed first.

**Comment by dkanada** [comment]
It also seems like the CI jobs are failing.

File: Emby.Notifications/Api/NotificationsService.cs (hunk 2 above)
Comment (Bond-009) [reply Emby.Notifications/Api/NotificationsService.cs]: I did

**Comment by JustAMan** [comment]
Can we first fix clients to _not_ use non-functioning API?

**Comment by Bond-009** [comment]
They are added back in

File: Emby.Notifications/Api/NotificationsService.cs [hunk 4]
```diff
@@ -127,32 +135,29 @@ public NotificationsService(INotificationManager notificationManager, IUserManag
...
         {
             return new NotificationsSummary
             {
-
             };
         }
 
         public Task Post(AddAdminNotification request)
         {
             // This endpoint really just exists as post of a real with sickbeard
```
Comment (JustAMan) [inline Emby.Notifications/Api/NotificationsService.cs]: wtf this comment means?

Comment (Bond-009) [reply Emby.Notifications/Api/NotificationsService.cs]: No idea

**Pull request merged**